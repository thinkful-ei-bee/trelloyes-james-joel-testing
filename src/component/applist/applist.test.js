import React from 'react';
import ReactDOM from 'react-dom';
import STORE from '../../store/store';
import renderer from 'react-test-renderer';

import AppList from './applist';

describe('<AppList />', () => {
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<AppList store={STORE}/>, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });
});

describe('<AppList />', () => {
    it('renders without crashing', () => {
        // Render the component, as JSON
        const tree = renderer.create(<AppList store={STORE}/>).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});