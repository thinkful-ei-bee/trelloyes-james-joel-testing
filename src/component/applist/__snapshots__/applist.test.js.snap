// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppList /> renders without crashing 1`] = `
<div
  className="App-list"
>
  <section
    className="List"
  >
    <header
      className="List-header"
    >
      <h2>
        First list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3>
          First card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Second card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Fifth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Sixth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Seventh card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Tenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Twelth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Thirteenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
    </div>
  </section>
  <section
    className="List"
  >
    <header
      className="List-header"
    >
      <h2>
        Second list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3>
          Second card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Third card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Fourth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Sixth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Eighth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Ninth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Eleventh card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
    </div>
  </section>
  <section
    className="List"
  >
    <header
      className="List-header"
    >
      <h2>
        Third list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3>
          First card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Second card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Third card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Fourth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Fifth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Sixth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Seventh card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Eighth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Ninth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Tenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Eleventh card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Twelth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Thirteenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
    </div>
  </section>
  <section
    className="List"
  >
    <header
      className="List-header"
    >
      <h2>
        Fourth list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3>
          Twelth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="Card"
      >
        <h3>
          Thirteenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
    </div>
  </section>
</div>
`;
